rs/node_rewards/canister/api/src/api_native_conversion.rs

```diff
@@ -0,0 +1,249 @@
+                node_metrics: Some(NodeMetricsDaily::from(node_metrics)),
+            },
+            native_types::NodeStatus::Unassigned { extrapolated_fr } => Self::Unassigned {
+                extrapolated_fr_percent: Some(DecimalProto::from(extrapolated_fr)),
+            },
+        }
+    }
+}
+
+impl TryFrom<NodeStatus> for native_types::NodeStatus {
+    type Error = String;
+
+    fn try_from(value: NodeStatus) -> Result<Self, Self::Error> {
+        match value {
+            NodeStatus::Assigned { node_metrics } => Ok(Self::Assigned {
+                node_metrics: native_types::NodeMetricsDaily::try_from(node_metrics.unwrap())?,
+            }),
+            NodeStatus::Unassigned {
+                extrapolated_fr_percent,
+            } => Ok(Self::Unassigned {
+                extrapolated_fr: Decimal::try_from(extrapolated_fr_percent.unwrap())?,
+            }),
+        }
+    }
+}
+
+// NodeResults conversions
+impl From<native_types::NodeResults> for NodeResults {
+    fn from(results: native_types::NodeResults) -> Self {
+        Self {
+            node_id: Some(results.node_id.get()),
+            node_reward_type: Some(results.node_reward_type.to_string()),
+            region: Some(results.region),
+            dc_id: Some(results.dc_id),
+            node_status: Some(NodeStatus::from(results.node_status)),
+            performance_multiplier_percent: Some(DecimalProto::from(
+                results.performance_multiplier,
+            )),
+            rewards_reduction_percent: Some(DecimalProto::from(results.rewards_reduction)),
+            base_rewards_xdr_permyriad: Some(DecimalProto::from(results.base_rewards)),
+            adjusted_rewards_xdr_permyriad: Some(DecimalProto::from(results.adjusted_rewards)),
+        }
+    }
+}
+
+impl TryFrom<NodeResults> for native_types::NodeResults {
+    type Error = String;
+    fn try_from(value: NodeResults) -> Result<Self, Self::Error> {
+        Ok(Self {
+            node_id: NodeId::from(value.node_id.unwrap()),
+            node_reward_type: NodeRewardType::from(value.node_reward_type.unwrap()),
+            region: value.region.unwrap(),
+            dc_id: value.dc_id.unwrap(),
+            node_status: native_types::NodeStatus::try_from(value.node_status.unwrap())?,
+            performance_multiplier: Decimal::try_from(
+                value.performance_multiplier_percent.unwrap(),
+            )?,
+            rewards_reduction: Decimal::try_from(value.rewards_reduction_percent.unwrap())?,
+            base_rewards: Decimal::try_from(value.base_rewards_xdr_permyriad.unwrap())?,
+            adjusted_rewards: Decimal::try_from(value.adjusted_rewards_xdr_permyriad.unwrap())?,
+        })
+    }
+}
+
+// BaseRewards conversions
+impl From<native_types::BaseRewards> for BaseRewards {
+    fn from(rewards: native_types::BaseRewards) -> Self {
+        Self {
+            monthly_xdr_permyriad: Some(DecimalProto::from(rewards.monthly)),
```

Comment:
This predates this PR (therefore, do not make changes in this PR on account of this issue), but...

If you use a Decimal type, why permyriad? Seems like the whole point of Decimal is that you can use the "natural" scale, i.e. 1 XDR, not 1e-4 XDR! The whole point of using cents, e8s, etc is so that you can stay in integer land. Whereas, Decimal frees you from such shackles.